modules_extra/network.js

```diff
@@ -100,77 +95,166 @@ function duration (s) {
     return round(s)+'ms'
 }
 
+function peerListSort (a, b) {
+  var states = {
+    connected: 3,
+    connecting: 2
+  }
+
+  //types of peers
+  var types = {
+    modern: 4,
+    legacy: 3,
+    inactive: 2,
+    unattempted: 1,
+    other: 0
+  }
+
+  return (
+    (states[b.state] || 0) - (states[a.state] || 0)
+    || origin(b) - origin(a)
+    || types[getType(b)] - types[getType(a)]
+    || b.stateChange - a.stateChange
+  )
+}
+
+function formatDate (time) {
+  return new Date(time).toString()
+}
 
+function humanDate (time) {
+  return human(new Date(time)).replace(/minute/, 'min').replace(/second/, 'sec')
+}
 
 exports.create = function (api) {
 
   return {
-    menu_items: function () {
-      return h('a', {href: '#/network'}, '/network')
-    },
-
-    builtin_tabs: function () {
-      return ['/network']
-    },
-
-    screen_view: function (path) {
-
-      if(path !== '/network') return
-
-      var ol = h('ul.network')
-
-      ;(function poll () {
-
-        //if this tab isn't open, don't update.
-        //todo: make a better way to do this...
-        if(!isVisible(ol))
-          return setTimeout(poll, 1000)
-
-        api.sbot_gossip_peers(function (err, list) {
-          ol.innerHTML = ''
-          list.sort(function (a, b) {
-            return (
-              (states[b.state] || 0) - (states[a.state] || 0)
-              || origin(b) - origin(a)
-              || types[getType(b)] - types[getType(a)]
-              || b.stateChange - a.stateChange
-            )
-          }).forEach(function (peer) {
-            ol.appendChild(h('div',
-              api.avatar(peer.key, 'thumbnail'),
-              h('div',
-                peer.state || 'not connected',
-                ' ',
-                getType(peer),
-                ' ',
-                //TODO: show nicer details, with labels. etc.
-                (peer.ping && peer.ping.rtt) ? duration(peer.ping.rtt.mean) : '',
-                ' ',
-                (peer.ping && peer.ping.skew) ? duration(peer.ping.skew.mean) : '',
-                h('label',
-                  {title: new Date(peer.stateChange).toString()},
-                  peer.stateChange && ('(' + human(new Date(peer.stateChange))) + ')')
-                ),
-                'source:'+peer.source,
-                h('pre', legacyToMultiServer(peer)),
-                h('button', 'connect', {onclick: function () {
-                  api.sbot_gossip_connect(peer, function (err) {
+    menu_items: () => h('a', {href: '#/network'}, '/network'),
+    builtin_tabs: () => ['/network'],
+    screen_view,
+    mcss: () => fs.readFileSync(__filename.replace(/js$/, 'mcss'), 'utf8')
+  }
+
+  function screen_view (path) {
+    if (path !== '/network') return
+
+    var peers = obs_gossip_peers(api)
+
+    return h('div', { style: {'overflow':'auto'}, className: 'column scroller' }, [
```

Comment:
this is copied code - it notice the scroller className, which we've not got to yet.
mmmmm maybe put this off till later ?